tool/tctl/common/autoupdate_command.go

```diff
@@ -0,0 +1,217 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/coreos/go-semver/semver"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/client/webclient"
+	autoupdatev1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/autoupdate/v1"
+	"github.com/gravitational/teleport/api/types/autoupdate"
+	"github.com/gravitational/teleport/lib/auth/authclient"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/teleport/lib/utils/interval"
+)
+
+// versionResponse is structure for formatting the autoupdate version response.
+type versionResponse struct {
+	TargetVersion string `json:"target_version"`
+}
+
+// AutoUpdateCommand implements the `tctl autoupdate` command for managing
+// autoupdate process for tools and agents.
+type AutoUpdateCommand struct {
+	app    *kingpin.Application
+	config *servicecfg.Config
+
+	updateCmd *kingpin.CmdClause
+	getCmd    *kingpin.CmdClause
+	watchCmd  *kingpin.CmdClause
+
+	mode               string
+	toolsTargetVersion string
+	proxy              string
+
+	// stdout allows to switch standard output source for resource command. Used in tests.
+	stdout io.Writer
+}
+
+// Initialize allows AutoUpdateCommand to plug itself into the CLI parser.
+func (c *AutoUpdateCommand) Initialize(app *kingpin.Application, config *servicecfg.Config) {
+	c.app = app
+	c.config = config
+	autoUpdateCmd := app.Command("autoupdate", "Teleport auto update commands.")
+
+	clientToolsCmd := autoUpdateCmd.Command("client-tools", "Client tools auto update commands.")
+
+	c.updateCmd = clientToolsCmd.Command("update", "Edit client tools auto update configuration.")
+	c.updateCmd.Flag("set-mode", `Sets the mode to enable or disable tools auto update in cluster.`).EnumVar(&c.mode, "enabled", "disabled")
+	c.updateCmd.Flag("set-target-version", `Defines client tools target version required to be updated.`).StringVar(&c.toolsTargetVersion)
+
+	c.getCmd = clientToolsCmd.Command("get", "Receive tools auto update target version.")
+	c.getCmd.Flag("proxy", `URL of the proxy`).StringVar(&c.proxy)
```

Comment:
do you need to ask the proxy URL? Can't we retrieve it from the cluster we are connected to?

Reply:
we request data from cluster if we don't specify the proxy, proxy flag needed to request data from any cluster without auth

Reply:
The command is intended to be used by unauthenticated systems that need to query the client tools version from the /webapi/find endpoint.

Reply:
RJ requested that we add this command to tctl to improve UX. I agree that `curl | jq` provides the same functionality, and I don't have a preference.